spawn/plugins/json_input_file.py

```diff
@@ -0,0 +1,70 @@
+# spawn
+# Copyright (C) 2018-2019, Simmovation Ltd.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301  USA
+"""Plugin for running tasks that take a single JSON input file as command line argument
+"""
+import json
+import logging
+
+from luigi import configuration
+
+from ..util.validation import validate_file, validate_dir
+from ..spawners import SingleInputFileSpawner
+from ..tasks import SimulationTask
+from ..simulation_inputs import JsonSimulationInput
+
+
+def create_spawner(task_exe, working_dir, base_file, runner_type):
```

Comment:
I haven't written a unit test for this. I don't know if you have any recommended ways of testing it

Reply:
For a unit test, you could:
- patch `configuration.get_config()` with a mock object and check that the configration parameters are set correctly
- check that the parameters you pass from the input file are accessible from the returned spawner